v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/transforms/MessageTransformation.java

```diff
@@ -0,0 +1,174 @@
+      String element = new String(pubsubMessage.getPayload());
+      messageId = pubsubMessage.getMessageId();
+      LOG.debug("PubSubMessage messageId: " + messageId);
+      LOG.debug("PubSubMessage payload: " + element);
+
+      if (pubsubMessage.getAttribute("key") != null) {
+        attributeKey = pubsubMessage.getAttribute("key");
+        LOG.debug("PubSubMessage attributeKey: " + attributeKey);
+        hashKey = hashKeyPrefix + attributeKey + namespaceDelimiter + messageId;
+      } else {
+        hashKey = hashKeyPrefix + messageId;
+      }
+      receiver.output(KV.of(hashKey, KV.of(fieldKey, element)));
+    }
+  }
+
+  public static class MessageToRedisLogs
+      extends DoFn<PubsubMessage, KV<String, KV<String, String>>> {
+    public boolean isValidJSON(String json) {
+      try {
+        objectMapper.readTree(json);
+      } catch (JacksonException e) {
+        return false;
+      }
+      return true;
+    }
+
+    @ProcessElement
+    public void processElement(
+        @Element PubsubMessage pubsubMessage,
+        OutputReceiver<KV<String, KV<String, String>>> receiver)
+        throws JsonProcessingException {
+      String element = new String(pubsubMessage.getPayload());
+      messageId = pubsubMessage.getMessageId();
+      LOG.debug("PubSubMessage messageId: " + messageId);
+      LOG.debug("PubSubMessage payload: " + element);
+
+      if (element.isEmpty() && element.isBlank()) {
+        return;
+      }
+
+      objectMapper.enable(DeserializationFeature.FAIL_ON_TRAILING_TOKENS);
+      if (!isValidJSON(element)) {
+        return;
+      }
+
+      Map<String, Object> elementMapData = objectMapper.readValue(element, Map.class);
+
+      if (elementMapData.containsKey("timestamp")) {
+        LocalDateTime localDateTime =
+            LocalDateTime.parse(
+                String.valueOf(elementMapData.get("timestamp")), DateTimeFormatter.ISO_DATE_TIME);
+        ZonedDateTime zonedDateTime = ZonedDateTime.of(localDateTime, ZoneId.systemDefault());
+        long timestampValueAsLong = zonedDateTime.toInstant().toEpochMilli();
+        elementMapData.put("timestampAsLong", String.valueOf(timestampValueAsLong));
+        hashKey = logKeyPrefix + timestampValueAsLong + namespaceDelimiter + messageId;
+      } else {
+        hashKey = logKeyPrefix + messageId;
+      }
+      elementMapData.forEach(
+          (k, v) -> receiver.output(KV.of(hashKey, KV.of(k, String.valueOf(v)))));
+    }
+  }
+
+  public static class MessageToRedisStreams
+      extends DoFn<PubsubMessage, KV<String, Map<String, String>>> {
+    @ProcessElement
+    public void processElement(
+        @Element PubsubMessage pubsubMessage,
+        OutputReceiver<KV<String, Map<String, String>>> receiver) {
+      String element = new String(pubsubMessage.getPayload());
+      messageId = pubsubMessage.getMessageId();
+      LOG.debug("PubSubMessage messageId: " + messageId);
+      LOG.debug("PubSubMessage payload: " + element);
```

Comment:
Same

Reply:
Removed payload debug and left the messageId.